packages/@react-spectrum/color/docs/ColorSlider.mdx

```diff
@@ -0,0 +1,222 @@
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License. -->
+
+import {Layout} from '@react-spectrum/docs';
+export default Layout;
+
+import docs from 'docs:@react-spectrum/color';
+import colorTypes from 'docs:@react-types/color/src/index.d.ts';
+import statelyDocs from 'docs:@react-stately/color';
+import {HeaderInfo, PropTable, TypeLink} from '@react-spectrum/docs';
+import packageData from '@react-spectrum/color/package.json';
+
+```jsx import
+import {ColorSlider} from '@react-spectrum/color';
+import {Flex} from '@react-spectrum/layout';
+```
+
+---
+category: Color
+keywords: [color slider]
+---
+
+# ColorSlider
+
+<p>{docs.exports.ColorSlider.description}</p>
+
+<HeaderInfo
+  packageData={packageData}
+  componentNames={['ColorSlider']}
+  sourceData={[
+    {type: 'Spectrum', url: 'https://spectrum.adobe.com/page/color-slider/'}
+  ]} />
+
+## Example
+
+```tsx example
+<ColorSlider defaultValue="#7f0000" channel="red" />
+```
+
+## Value
+
+A ColorSlider requires either a uncontrolled default value or a controlled value, provided using the `defaultValue` or `value` props respectively.
+The value provided to either of these props should be a string or <TypeLink links={colorTypes.links} type={colorTypes.exports.Color} />  object that matches the provided ColorSlider's `channel` prop. For example, a
+ColorSlider representing a color channel ("red", "green", "blue") would have a hex color code (e.g. "#7f0000") provided. For a full list of supported channels,
+see the [Props](#props) table below.
+
+```tsx example
+function Example() {
+  let [value, setValue] = React.useState('hsl(0, 100%, 50%)');
+  return (
+    <Flex gap="size-150" wrap>
+      <ColorSlider
+        defaultValue="hsl(0, 100%, 50%)"
+        channel="hue" />
+      <ColorSlider
+        value={value}
+        onChange={setValue}
+        channel="hue" />
+    </Flex>
+  );
+}
+```
+
+## Labeling
+
+The ColorSlider's channel and value are shown above the ColorSlider by default. The label and value label can be hidden by providing
+`label={null}` and `showValueLabel={false}` to the ColorSlider.
+
+```tsx example
+<Flex gap="size-150" wrap>
+  <ColorSlider channel="saturation" defaultValue="hsl(0, 100%, 50%)" label={null} />
+  <ColorSlider channel="lightness" defaultValue="hsl(0, 100%, 50%)" showValueLabel={false} />
+</Flex>
+```
+
+### Accessibility
+
+If the ColorSlider's label is hidden, a localized string for the channel name is used as the `aria-label` by default. The value of the ColorSlider
```

Comment:
If we are populating the label by default, when it is null should be populated aria-label by default?

Reply:
yep